src/Tribe/Generator/Event.php

```diff
@@ -109,13 +114,158 @@ public function random_event_data( $fromDate, $toDate, $is_virtual ) {
                     '_tribe_events_zoom_join_url'               => 'https:\/\/zoom.us\/j\/1100000',
                     '_tribe_events_zoom_join_instructions'      => 'https:\/\/support.zoom.us\/hc\/en-us\/articles\/201362193-Joining-a-Meeting', 
                     '_tribe_events_zoom_global_dial_in_numbers' => ['+1 1000200200' => 'US', '+1 6699006833' => 'US']
-                ];
+                ] );
             }
+        }
+
+		if( $is_recurring ) {
+		    //Force recurring events creation in foreground
+            $lots_of_them = static function () {
+                return PHP_INT_MAX;
+            };
+            add_filter( 'tribe_events_pro_recurrence_small_batch_size', $lots_of_them );
+            add_filter( 'tribe_events_pro_recurrence_batch_size', $lots_of_them );
+
+		    $type = $this->get_recurrence_type( $recurring_type );
+		    $count = $this->get_recurrence_count( $type );
+		    $recurrence_date = Dates::build_date_object( $event_date['start'] );
+
+		    switch( $type ) {
+
+                case 'Weekly':
```

Comment:
There is a simpler way to do this - start with your default and then check for 'Weekly" or 'Yearly" and add to the array. Before doing the `array_merge()`.

Reply:
We can definitely simplify this, for sure!